modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/ExtensionModelParser.java

```diff
@@ -103,4 +105,14 @@
    */
   List<ModelProperty> getAdditionalModelProperties();
 
+  /**
+   * @return the extension's {@link DeprecationModel} if one was defined
+   */
+  Optional<DeprecationModel> getDeprecationModel();
+
+
+  /**
+   * @return the extension's {@link XmlDslModel}
+   */
+  XmlDslModel getXmlDslModel();
```

Comment:
I'm thinking if this method should provide only the information that the annotation provides instead, and the loader would create the XmlDslModel then.